context/context.go

```diff
@@ -181,15 +192,22 @@ type Response struct {
 // started means the response has sent out.
 func (w *Response) Write(p []byte) (int, error) {
 	w.Started = true
+	if !w.wroteHeader {
+		w.ResponseWriter.WriteHeader(w.Status)
+		//prevent multiple response.WriteHeader calls
+		w.wroteHeader = true
+	}
 	return w.ResponseWriter.Write(p)
 }
 
 // WriteHeader sends an HTTP response header with status code,
 // and sets `started` to true.
 func (w *Response) WriteHeader(code int) {
+	if w.Status > 0 {
+		return
+	}
 	w.Status = code
 	w.Started = true
-	w.ResponseWriter.WriteHeader(code)
```

Comment:
I think we just leave this `w.ResponseWriter.WriteHeader(code)`. lost some header is acceptable. As they know set header should before set status.